src/renderer/sw_engine/tvgSwFill.cpp

```diff
@@ -125,15 +125,24 @@ static void _applyAA(const SwFill* fill, uint32_t begin, uint32_t end)
 
 static bool _updateColorTable(SwFill* fill, const Fill* fdata, const SwSurface* surface, uint8_t opacity)
 {
+    const Fill::ColorStop* colors;
```

Comment:
We can move this part to: (see 1923)
```
Fill::ColorStop* FillFetchSolid(...);
```

Reply:
to do this I need to have access to `const Fill* fdata` inside `FillFetchSolid` fun (and also to the opacity). seems a bit strange to have access to this in tvgRaster, doesn't it? 
instead I can pass const Fill::ColorStop* and the size to the `rasterGradientShape` but then `fillFetchSolid` seems to be useless, and also the same question, whether passing them to tvgRaster is ok?

Reply:
@mgrudzinska

> to do this I need to have access to `const Fill* fdata` inside `FillFetchSolid` fun (and also to the opacity). seems a bit strange to have access to this in tvgRaster, doesn't it?

If necessary, we can do it. The engine internals won't be reused other places and are inevitably already tightly coupled. 

> I can pass const Fill::ColorStop* and the size to the `rasterGradientShape` but then `fillFetchSolid` seems to be useless,

Maybe. I'm not sure whether i got your point. 

Please consider this style again: with no need to introduce an unlikely used `color` member.

```
if (auto color = FillFetchSolid(shape->fill)) {
   return raster...(c->r, c->g, c->b, MULTIPLY(c->a, opacity));
}
```